### PR TITLE
[CI:BUILD] copr: fix el8 builds

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

#### What type of PR is this?

/kind other

#### What this PR does / why we need it:

Fixes copr build

#### How to verify it

successful el8 build at: https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/build/5072942/

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None
#### Does this PR introduce a user-facing change?

```release-note
None
```

@flouthoc @nalind @rhatdan PTAL

### CODE REVIEW

**Comment by lsm5** [comment]
older builds were failing for el8: https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/build/5072893/

**Comment by rhatdan** [comment]
@lsm5 you still working on this?

**Comment by lsm5** [comment]
> @lsm5 you still working on this?

I'll resume this, been working on a similar one for skopeo.

**Comment by vrothberg** [comment]
> Why doesn't the next `%if 0%{?rhel}` block ~not~ accomplish the same thing? (Aside: the use of single quotes in that next block appears to set BUILDTAGS to a value that literally includes "$BUILDTAGS", rather than expanding it.) From the build log, it looks like `$BUILDTAGS` is either empty or not being referenced when `go build` is run, which looks like a bigger problem than the current patch would work around.

This looks still open

**Comment by lsm5** [comment]
err yes, I still gotta work on this :) . Let me try to get skopeo out of the way first. If we prefer closing this and reopening when I'm ready to update, fine by me.

**Comment by lsm5** [comment]
RHEL8's default %gobuild didn't account for $BUILDTAGS. Adjusted that in buildah.spec.rpkg . Successful build at https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/build/5390214/ .

**Comment by lsm5** [comment]
let me fix debuginfo in this PR itself.

**Comment by lsm5** [comment]
ready for review. successful build at https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/build/5406349/

**Review by flouthoc — APPROVED** [review]
LGTM

**Comment by lsm5** [comment]
i'll update this PR further with making the spec as close to podman and buildah wherever possible.

File: buildah.spec.rpkg [hunk 1]
```diff
@@ -11,17 +11,19 @@
 
 %global with_debug 1
 
+# RHEL 8's default %%gobuild macro doesn't account for the BUILDTAGS variable, so we
```
Comment (TomSweeneyRedHat) [inline buildah.spec.rpkg:14]: Soft suggestion:
```suggestion
# Starting with RHEL 8, the default %%gobuild macro doesn't account for the BUILDTAGS variable, so we
```

File: buildah.spec.rpkg [hunk 2]
```diff
@@ -11,17 +11,19 @@
...
+# set it separately here and do not depend on RHEL 8's go-srpm-macros package.
```
Comment (TomSweeneyRedHat) [inline buildah.spec.rpkg:15]: If you take the prior
```suggestion
# set it separately here and do not depend on RHEL's go-srpm-macros package.
```

Comment (lsm5) [reply buildah.spec.rpkg:15]: @TomSweeneyRedHat nope, this is specifically for rhel8. rhel 9 is fine.

File: buildah.spec.rpkg (hunk 1 above)
Comment (lsm5) [reply buildah.spec.rpkg:14]: Not quite `starting with rhel8`. But only rhel8.

**Comment by lsm5** [comment]
good to go. PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**